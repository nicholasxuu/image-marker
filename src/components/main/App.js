/* eslint-disable react/prefer-stateless-function,arrow-body-style */
import React from 'react';
import PropTypes from 'prop-types';
import queryParam from 'query-string';
import ImageMarker from '../../containers/ImageMarkerContainer';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      imageUrl: '',
      initialList: null,
      disabled: false,
    };
  }

  componentDidMount = () => {
    const { search } = this.props.location;
    if (search) {
      const query = queryParam.parse(search);
      const newState = {};
      if (query.imageurl) {
        newState.imageUrl = query.imageurl;
      }

      if (query.initlist) {
        newState.initialList = JSON.parse(query.initlist);
      }

      if (query.disabled) {
        newState.disabled = (query.disabled === '1');
      }

      this.setState(newState);
    }
  };

  render = () => {
    return (
      <ImageMarker
        imageUrl={this.state.imageUrl}
        initialList={this.state.initialList}
        disabled={this.state.disabled}
      />
    );
  }
}

App.defaultProps = {
  location: {
    search: null,
  },
};

App.propTypes = {
  location: PropTypes.shape({
    search: PropTypes.string,
  }),
};

export default App;

