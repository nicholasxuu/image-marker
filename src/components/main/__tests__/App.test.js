/* eslint-disable function-paren-newline,prefer-const */
// Link.react-test.js
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

test('sample component snapshot', () => {
  const component = renderer.create(
    <App />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
