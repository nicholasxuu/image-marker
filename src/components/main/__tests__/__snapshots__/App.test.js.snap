// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sample component snapshot 1`] = `
<div>
  hello world
</div>
`;
